proposed/runtime-binding.md

```diff
@@ -229,37 +225,80 @@ Can be specified via the following ways:
+The following example shows the existing behavior of this knob:
 
 ```console
-C:\testapps\twotwoapp>dotnet --fx-version 3.0.0 bin\Debug\netcoreapp2.2\twotwoapp.dll
+C:\testapps\twooneapp>type Program.cs
+using System;
+
+namespace twooneapp
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Hello World!");
+            Console.WriteLine(typeof(Object).Assembly.Location);
+        }
+    }
+}
+C:\testapps\twooneapp>
+C:\testapps\twooneapp>dotnet bin\Debug\netcoreapp2.1\twooneapp.dll
+Hello World!
+C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.1.7\System.Private.CoreLib.dll
+
+C:\testapps\twooneapp>dotnet --fx-version 2.1.0 bin\Debug\netcoreapp2.1\twooneapp.dll
+Hello World!
+C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.1.0\System.Private.CoreLib.dll
 ```
 
-The example above will attempt to load the 3.0.0 runtime version. If it isn't found, the host will print an error message.
+The `--fx-version` argument can be composed with the new `RollForward` settings, either via CLI or ENV. By default, `--fx-version` does not roll-forward.
 
-Another example:
+The following example enables a .NET Core 2.1 application to run on .NET Core 2.2 without having to select the correct 2.2 patch version. In this case `--fx-version` specifies the floor version. In this scenario, 2.2.0 is not on the machine, but a later patch version is.
 
 ```console
-C:\testapps\threezeroapp>bin\Debug\netcoreapp3.0\threezero.exe --fx-version 3.0.0 --roll-forward patch
+C:\testapps\twooneapp>dotnet --fx-version 2.2.0 bin\Debug\netcoreapp2.1\twooneapp.dll
+It was not possible to find any compatible framework version
+The specified framework 'Microsoft.NETCore.App', version '2.2.0' was not found.
+C:\testapps\twooneapp>dotnet --fx-version 2.2.0 --roll-forward Patch bin\Debug\netcoreapp2.1\twooneapp.dll
+Hello World!
+C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.2.3\System.Private.CoreLib.dll
 ```
 
-The example above will attempt to load the highest `3.0` patch version, which may be `3.0.0` or `3.0.27`, for example, whichever is the highest path version on the machine.
-
 ### Configuration Precedence
 
-The host will consult these settings in order (later scopes take precendence over earlier ones):
+The host will consult the various ways of setting `RollForward`, in order (later scopes take precendence over earlier ones):
+
+1. `.runtimeconfig.json` properties (AKA "json")
+2. CLI arguments
+3. Environment variables (AKA "ENV")
 
-* `.runtimeconfig.json` properties (AKA "json")
-* CLI arguments
-* Environment variables (AKA "ENV")
+Note: The project file property is not listed because the project file is a build-time not runtime artifact. Runtime-oriented values in project files need to be written to `.runtimeconfig.json` as part of the build in order to influence runtime behavior.
 
 The `version` and `roll-forward` settings compose in the following way:
```

Comment:
It would be good to list the various ways to specify `version`. Is it the version of framework in `runtimeconfig` and `--fx-version` CLI, or is there something else as well?

Reply:
I am only aware of those two and cannot think of a useful other option. We could have an ENV to specify version, but I suspect you'd largely want that to test latest and we now have an ENV that does that.

Reply:
OK - those two make sense - maybe we should list them here, as they are not explicitly mentioned together anywhere in this doc I think...